pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -500,21 +522,22 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 	// Render the desired objects from the CRD and create or update them.
 	hasNoLicense := !utils.IsFeatureActive(license, common.ThreatDefenseFeature)
 	intrusionDetectionCfg := &render.IntrusionDetectionConfiguration{
-		IntrusionDetection:    *instance,
-		LogCollector:          lc,
-		ESSecrets:             esSecrets,
-		Installation:          network,
-		ESClusterConfig:       esClusterConfig,
-		PullSecrets:           pullSecrets,
-		Openshift:             r.provider == operatorv1.ProviderOpenShift,
-		ClusterDomain:         r.clusterDomain,
-		ESLicenseType:         esLicenseType,
-		ManagedCluster:        isManagedCluster,
-		ShouldRenderADPVC:     shouldRenderADPVC,
-		HasNoLicense:          hasNoLicense,
-		TrustedCertBundle:     trustedBundle,
-		ADAPIServerCertSecret: adAPIServerTLSSecret,
-		UsePSP:                r.usePSP,
+		IntrusionDetection:           *instance,
+		LogCollector:                 lc,
+		ESSecrets:                    esSecrets,
+		Installation:                 network,
+		ESClusterConfig:              esClusterConfig,
+		PullSecrets:                  pullSecrets,
+		Openshift:                    r.provider == operatorv1.ProviderOpenShift,
+		ClusterDomain:                r.clusterDomain,
+		ESLicenseType:                esLicenseType,
+		ManagedCluster:               isManagedCluster,
+		ShouldRenderADPVC:            shouldRenderADPVC,
+		HasNoLicense:                 hasNoLicense,
+		TrustedCertBundle:            trustedBundle,
+		ADAPIServerCertSecret:        adAPIServerTLSSecret,
+		IntrusionDetectionCertSecret: keyPair,
```

Comment:
This should be `IntrusionDetectionKeyPair` - it's slightly different from the `XCertSecret` vars, which only include the public certs of the other components.